scripts/staging_webhook.py

```
@@ -0,0 +1,241 @@
+        return None
+
+
+def fetch_pr_data(repo: str, pr_number: int, token: Optional[str] = None) -> dict:
+    """Fetch PR data from GitHub API."""
+    import urllib.request
+    import urllib.error
+
+    url = f"https://api.github.com/repos/{repo}/pulls/{pr_number}"
+    headers = {
+        "Accept": "application/vnd.github.v3+json",
+        "User-Agent": "staging-webhook-cli"
+    }
+    if token:
+        headers["Authorization"] = f"token {token}"
+
+    request = urllib.request.Request(url, headers=headers)
+
+    try:
+        with urllib.request.urlopen(request, timeout=30) as response:
+            return json.loads(response.read().decode())
+    except urllib.error.HTTPError as e:
+        if e.code == 404:
+            raise ValueError(f"PR #{pr_number} not found in {repo}")
+        elif e.code == 401:
+            raise ValueError("GitHub authentication failed. Set GITHUB_TOKEN or run 'gh auth login'")
+        else:
+            raise ValueError(f"GitHub API error: {e.code} {e.reason}")
+    except urllib.error.URLError as e:
+        raise ValueError(f"Network error: {e.reason}")
+
+
+def construct_task_payload(pr_data: dict, repo: str) -> dict:
+    """Construct a task payload matching the orchestrator format."""
+    pr_number = pr_data["number"]
+    pr_title = pr_data["title"]
+    pr_url = pr_data["html_url"]
+    head_sha = pr_data["head"]["sha"]
+
+    task_id = f"staging-{uuid.uuid4().hex[:8]}-{pr_number}"
+
+    question = f"Review PR #{pr_number}: {pr_title}"
+
+    context = {
+        "resource_id": pr_number,
+        "resource_type": "pull_request",
+        "url": pr_url,
+        "head_sha": head_sha,
+        "source": "staging_webhook_cli",
+        "triggered_at": datetime.now(timezone.utc).isoformat(),
+    }
+
+    return {
+        "task_id": task_id,
+        "question": question,
+        "repo": repo,
+        "task_type": "review",
+        "context": context,
+    }
```

IMPACT: This payload structure must exactly match the orchestrator's expected schema. Any mismatch (e.g., missing fields, type differences) will cause silent failures or crashes in downstream services. There is no versioning or validation here, so changes in production can break staging simulation without detection. SUGGESTED FIX: Add a schema version field (e.g., 'version': 1) and validate against a shared contract/schema if available. Alternatively, reference a common payload model instead of reconstructing manually.